### ISSUE TITLE
CPU Meter issue

### ISSUE THREAD

**Issue by Shrekmachine** [raiguard/ModernGadgets#119]
CPU Meter creating many errors liekt his: "Formula: Division by 0 in key "Scale69" in [MeterLineGraph] (ModernGadgets\CPU\CPU.ini)"
latest Version 1.4.1

**Comment by Shrekmachine** [comment]
thanks, that helped. Plus, that updating HIW SMV is not updating the Skin. I have to refresh "manually"

**Comment by raiguard** [comment]
Yeah, that's normal. You can click the refresh button next to the blue book icon above the SMV button to refresh the skin.

**Comment by Shrekmachine** [comment]
I would like to keep it open. When I add a new value from HWInfo to Core 0, I get a value displayed for Core 0-4. Rest not.
AMD 2700x with GB Aouros Ultra Gaming 
![image](https://user-images.githubusercontent.com/8019511/47592924-a9532680-d975-11e8-9e0b-d9838ed272ed.png)

**Comment by raiguard** [comment]
Did you open the `AMD Ryzen 7 2700X: Enhanced` section and get the actual entry from there? If you set it to what you currently have highlighted, then it's not working because it's configured incorrectly.

**Comment by Shrekmachine** [comment]
this one I took:
![image](https://user-images.githubusercontent.com/8019511/47601390-10052e00-d9d0-11e8-9046-b075f71ac674.png)

**Comment by Shrekmachine** [comment]
update: and core 11/12 :o

**Comment by raiguard** [comment]
So I think I know why this is happening. By default, the values after zero are set to take the same sensor ID and instance that are set for core 0, but then they have sequential entry IDs. So cores 1-4 work because there is a sensor that uses the same sensor ID and instance as the core 0 entry, but has 0x10000001 as the entry ID. When you get to core 11, it used 0x1000000a as the entry ID.

Anyway, what I am trying to say is that this is the consequence of an intended feature. However, it appears that I was right in that your CPU only has one temperature sensor, and doesn't have one for each core. So you're out of luck for now. :(

I am working on an alternate temperature display that displays the overall CPU temperature rather than core temperatures, so be on the lookout for that in the near future.

**Comment by Shrekmachine** [comment]
ok, thanks. I scanned all values in SMV, there is only CPU (Tctl) and (Tdie). not a set of values per core. 
I leave it for now as it is displayed.
Interestingly: When I start core temp, all cores showed the same temparature as long as I did not touch the config.

**Comment by raiguard** [comment]
Yeah, so that's what's happening. Sorry about that! I'll work on the new display today so hopefully it'll be in the next v1.5 beta.

**Comment by Shrekmachine** [comment]
great support, thank you. Suggestion: For AMD CPU's remove the temp from the individual core lines. Implementing it like e.g. Fan would be nice. 
or better show the min on the left, the max on the right and the current as a slider. 
Same for the fan would be cool. 
You could call it "min/max dynamic" mode :)

**Comment by Shrekmachine** [comment]
or fan / Temp / Volt /Clock in 1 line to minimize consumed space

**Comment by raiguard** [comment]
Yeah, I plan on adding it like the dynamic fan display. I think it's pretty useless to show min/max info though, because the min is obviously zero, and you're likely to know the max (for example, 99% of CPUs have a maximum temperature of 100 degrees Celsius).

The bad news is that when I started to implement this new display, I discovered that I made a huge mistake when implementing the SpeedFan core temperatures: Nowhere in the code does it set the automatic maximum. That causes the same issue that you originally reported (the divide by zero error). So I'll have to fix that before I can get to the new display.

**Comment by raiguard** [comment]
![2018-10-27 19_40_58-d__settings_caleb_rainmeter_skins_moderngadgets_cpumeter_cpumeter ini](https://user-images.githubusercontent.com/3515394/47610980-436fa900-da20-11e8-866e-1d96e2b8e517.png)

**Comment by raiguard** [comment]
![cputemp](https://user-images.githubusercontent.com/3515394/47611138-cdba0c00-da24-11e8-8fdd-89d3cbfd5f08.gif)

**Comment by Shrekmachine** [comment]
after manually injecting the CPU Meter files yesterday it looks like being stable :) I just had to adjust the Core Temp assignment in SMV

**Comment by raiguard** [comment]
Ah, yeah, sorry, I've been busy with life (just got a promotion at work!), so I didn't have a chance to work on things.

Just FYI, the 'standard' display doesn't use the Core temperature values, it uses the package temperature value in the 'general' section.

**Comment by raiguard** [comment]
May I ask, can you find a CPU voltage reading somewhere in HWiNFO? I was considering adding a standard voltage display as well but my CPU doesn't seem to have an overall voltage value.

**Comment by Shrekmachine** [comment]
With me the voltage is displayed out of the box if I tick the voltage checkbox. 
![image](https://user-images.githubusercontent.com/8019511/47727036-fb3cca80-dc5b-11e8-81b4-128158dbd4cf.png)

For the temp indeed I had to assign the CPU value to the package temp
![image](https://user-images.githubusercontent.com/8019511/47727063-07288c80-dc5c-11e8-819c-5f9c0618916d.png)

**Comment by Shrekmachine** [comment]
total Voltage 
![image](https://user-images.githubusercontent.com/8019511/47733814-5b863900-dc69-11e8-9d00-15a8333614c3.png)

**Comment by raiguard** [comment]
Yeah, I guess I'll go ahead and add a 'standard' display, but my CPU doesn't have a general voltage value like yours does, so it'll be very situational.

**Comment by Shrekmachine** [comment]
let me know in case you need more AMD specific values or want me to test something
